src/pyscipopt/expr.pxi

```diff
@@ -247,3 +343,324 @@ def quicksum(termlist):
+        if divisor.getOp() == Operator.const:
+            assert divisor.number != 0.0
+
+        return self * divisor**(-1)
+
+    def __neg__(self):
+        return -1.0 * self
+
+    def __sub__(self, other):
+        return self + (-other)
+
+    def __radd__(self, other):
+        return self.__add__(other)
+
+    def __rmul__(self, other):
+        return self.__mul__(other)
+
+    def __rsub__(self, other):
+        return -1.0 * self + other
+
+    def __richcmp__(self, other, op):
+        '''turn it into a constraint'''
+        return _expr_richcmp(self, other, op)
+
+    def degree(self):
+        return float('inf') # none of these expressions should be polynomial
+
+    def getOp(self):
+        return self.op
+
+
+# Sum Expressions
+cdef class SumExpr(GenExpr):
+
+    cdef public constant
+    cdef public coefs
+
+    def __init__(self):
+        self.constant = 0.0
+        self.coefs = []
+        self.children = []
+        self.op = Operator.add
+        self.operatorIndex = Operator.operatorIndexDic[self.op]
+    def __repr__(self):
+        return self.op + "(" + str(self.constant) + "," + ",".join(map(lambda child : child.__repr__(), self.children)) + ")"
+
+# Prod Expressions
+cdef class ProdExpr(GenExpr):
+    cdef public constant
+    def __init__(self):
+        self.constant = 1.0
+        self.children = []
+        self.op = Operator.prod
+        self.operatorIndex = Operator.operatorIndexDic[self.op]
+    def __repr__(self):
+        return self.op + "(" + str(self.constant) + "," + ",".join(map(lambda child : child.__repr__(), self.children)) + ")"
+
+# Var Expressions
+cdef class VarExpr(GenExpr):
+    cdef public var
+    def __init__(self, var):
+        self.children = [var]
+        self.op = Operator.varidx
+        self.operatorIndex = Operator.operatorIndexDic[self.op]
+    def __repr__(self):
+        return self.children[0].__repr__()
+
+# Pow Expressions
+cdef class PowExpr(GenExpr):
+    cdef public expo
+    def __init__(self):
+        self.expo = 1.0
+        self.children = []
+        self.op = Operator.power
+        self.operatorIndex = Operator.operatorIndexDic[self.op]
+    def __repr__(self):
+        return self.op + "(" + self.children[0].__repr__() + "," + str(self.expo) + ")"
+
+# Exp, Log, Sqrt Expressions
+cdef class UnaryExpr(GenExpr):
+    def __init__(self):
+        self.children = []
+    def __repr__(self):
+        return self.op + "(" + self.children[0].__repr__() + ")"
+
+# class for constant expressions
+cdef class Constant(GenExpr):
+    cdef public number
+    def __init__(self,number):
+        self.number = number
+        self.op = Operator.const
+        self.operatorIndex = Operator.operatorIndexDic[self.op]
+
+    def __repr__(self):
+        return str(self.number)
+
+def exp(expr):
+    ans = UnaryExpr()
+    ans.op = Operator.exp
+    ans.operatorIndex = Operator.operatorIndexDic[ans.op]
+    ans.children.append(buildGenExprObj(expr))
+    return ans
```

Comment:
why do set all these attributes after creating an "empty" object, instead of using the `UnaryExpr.__init__` method to initialize them?

Reply:
what you suggest is way more elegant, will implement after lunch ;)

Reply:
Good, so I made one constructive suggestions ;-)